src/main/scala/viper/gobra/frontend/info/implementation/typing/ExprTyping.scala

```diff
@@ -275,6 +275,29 @@ trait ExprTyping extends BaseTyping { this: TypeInfoImpl =>
       }
     }
 
+    case _: PNew => noMessages
+
+    case m@PMake(typ, args) =>
+      args.flatMap { arg =>
+        assignableTo.errors(exprType(arg), INT_TYPE)(arg) ++
+          error(arg, s"arguments to make must be non-negative", intConstantEval(arg).exists(_ < 0))
+      } ++ (typ match {
+        case _: PSliceType =>
+          error(m, s"too many arguments to make($typ)", args.length > 2) ++
+            error(m, s"missing len argument to make($typ)", args.isEmpty) ++
+            check(args){
+              case args if args.length == 2 =>
+                val maybeLen = intConstantEval(args(0))
+                val maybeCap = intConstantEval(args(1))
+                error(m, s"len larger than cap in make($typ)", maybeLen.isDefined && maybeCap.isDefined && maybeLen.get > maybeCap.get)
+            }
+
+        case _: PBiChannelType | _: PMapType =>
```

Comment:
Isn't it fine to use `PSendChannelType` as well as `PRecvChannelType`?

Reply:
I was under the impression that you could not make send and receive channels for some reason. I still think that it does not make sense to make a channel of these types but I tried`make` with both types and it worked to my surprise. @Felalolf I am sorry if I induced you in error about this.

PS: I found a Github discussion about this same issue(https://github.com/golang/go/issues/2431)

Reply:
I will change this in my next commit

Reply:
I would go with a better error message in case someone tries to create a send or receive only channel but disallowing them seems fine to me